pkg/agent/util/net_windows.go

```diff
@@ -896,12 +898,186 @@ func VirtualAdapterName(name string) string {
+			return err
+		}
+	}
+
+	enabled, err := isOVSExtensionEnabled()
+	if err != nil {
+		return err
+	}
+	if !enabled {
+		if err = enableOVSExtension(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func RemoveVMSwitch() error {
+	exists, err := VMSwitchExists()
+	if err != nil {
+		return err
+	}
+	if exists {
+		cmd := fmt.Sprintf(`Remove-VMSwitch -Name "%s" -ComputerName $(hostname) -Force`, LocalVMSwitch)
+		_, err = ps.RunCommand(cmd)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func GenHostInterfaceName(upLinkIfName string) string {
+	return strings.TrimSuffix(upLinkIfName, bridgedUplinkSuffix)
+}
+
+// createVMSwitchWithTeaming creates VMSwitch and enables OVS extension.
+// Connection to VM is lost for few seconds
+func createVMSwitchWithTeaming(switchName, ifName string) error {
+	cmd := fmt.Sprintf(`New-VMSwitch -Name "%s" -NetAdapterName "%s" -EnableEmbeddedTeaming $true -AllowManagementOS $true -ComputerName $(hostname)| Enable-VMSwitchExtension "%s"`, switchName, ifName, ovsExtensionName)
+	_, err := ps.RunCommand(cmd)
+	if err != nil {
+		return err
+	}
 	return nil
 }
+
+func enableOVSExtension() error {
+	cmd := fmt.Sprintf(`Get-VMSwitch -Name "%s" -ComputerName $(hostname)| Enable-VMSwitchExtension "%s"`, LocalVMSwitch, ovsExtensionName)
+	_, err := ps.RunCommand(cmd)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func getRoutesOnInterface(linkIndex int) ([]interface{}, error) {
+	cmd := fmt.Sprintf("Get-NetRoute -InterfaceIndex %d -ErrorAction Ignore | Format-Table -HideTableHeaders", linkIndex)
+	rs, err := getNetRoutes(cmd)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get routes: %v", err)
+	}
+	var routes []interface{}
+	for _, r := range rs {
+		// Skip the routes automatically generated by Windows host when adding IP address on the network adapter.
+		if r.GatewayAddress != nil && r.GatewayAddress.IsUnspecified() {
+			continue
+		}
+		routes = append(routes, r)
+	}
+	return routes, nil
+}
+
+// parseOVSExtensionOutput parses the VM extension output
+// and returns the value of Enabled field
+func parseOVSExtensionOutput(s string) bool {
+	scanner := bufio.NewScanner(strings.NewReader(s))
+	for scanner.Scan() {
+		temp := strings.Fields(scanner.Text())
+		line := strings.Join(temp, "")
+		if strings.Contains(line, "Enabled") {
+			if strings.Contains(line, "True") {
+				return true
+			}
+			return false
+		}
+	}
+	return false
+}
+
+func isOVSExtensionEnabled() (bool, error) {
+	cmd := fmt.Sprintf(`Get-VMSwitchExtension -VMSwitchName "%s" -ComputerName $(hostname) | ? Id -EQ "%s"`, LocalVMSwitch, OVSExtensionID)
+	out, err := ps.RunCommand(cmd)
+	if err != nil {
+		return false, err
+	}
+	if strings.Contains(out, ovsExtensionName) {
+		ok := parseOVSExtensionOutput(out)
+		if !ok {
+			return false, nil
+		}
+		return true, nil
+	}
+	return false, fmt.Errorf("open vswitch extension driver is not installed")
```

Comment:
Sorry for being nit-picking: normal path could be indented least in the following way.
```suggestion
	if !strings.Contains(out, ovsExtensionName) {
	        return false, fmt.Errorf("open vswitch extension driver is not installed")
	}        
        return parseOVSExtensionOutput(out), nil
	
```

Reply:
Sure.